File: Handler/app/src/main/java/org/techtown/samplethread/MainActivity.java
package org.techtown.samplethread4;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    int value = 0;
    TextView textView;
    MainHandler mainHandler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textView);

        Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                BackgroundThread thread = new BackgroundThread();
                thread.start();
            }
        });

        mainHandler = new MainHandler();
    }

    class BackgroundThread extends Thread {
        public void run() {
            for (int i = 0 ; i <100 ; i++) {
                try{
                    Thread.sleep(1000);
                }catch (Exception e){}

                value += 1;
                Log.d("Thread","value : " + value);

                Message message = mainHandler.obtainMessage();
                Bundle bundle = new Bundle();
                bundle.putInt("value",value);
                message.setData(bundle);

                mainHandler.sendMessage(message);
            }
        }
    }
    
    class MainHandler extends Handler{
        @Override
        public void handleMessage(@NonNull Message msg) {
            super.handleMessage(msg);

            Bundle bundle = msg.getData();
            int value = bundle.getInt("value");
            textView.setText("value 값 : " + value);
        }
    }
}

File: Handler_kt/app/src/main/java/org/techtown/handler_kt/MainActivity.kt
package org.techtown.handler_kt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*
import java.lang.Exception

class MainActivity : AppCompatActivity() {

    var handler = Handler()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button.setOnClickListener {
            var thread = BackgroundThread()
            thread.start()
        }
    }

    inner class BackgroundThread : Thread() {
        var value : Int = 0

        public override fun run() : Unit {
            for(i in 1 until 100){
                try{
                    Thread.sleep(1000)
                }catch (e : Exception){}

                value += 1
                Log.d("Thread","value : " + value)

                handler.post(Runnable {
                    textView.text = "value 값 : ${value}"
                })
            }
        }
    }
}
